Framework/Algorithms/src/GetDetectorOffsets.cpp

```diff
@@ -164,21 +221,116 @@ void GetDetectorOffsets::exec() {
     childAlg->setPropertyValue("Filename", filename);
     childAlg->executeAsChildAlg();
   }
+
+  // Return the output fit result table
+  if (fit_result_table) {
+    std::string fit_result_table_name = getPropertyValue("OutputWorkspace");
+    fit_result_table_name += "_FitResult";
+    declareProperty(
+        make_unique<WorkspaceProperty<TableWorkspace>>(
+            "PeakFitResultTableWorkspace", fit_result_table_name,
+            Direction::Output),
+        "Name of the input Tableworkspace containing peak fit window "
+        "information for each spectrum. ");
+    setProperty("PeakFitResultTableWorkspace", fit_result_table);
+  }
+}
+
+/** Fit the peak for the second time
+ * @brief GetDetectorOffsets::fitPeakSecondTime
+ * @param wi
+ * @param isAbsolute
+ * @param minimum_peak_height
+ * @param fit_result
+ * @param mask_it
+ * @return
+ */
+double GetDetectorOffsets::fitPeakSecondTime(
+    size_t wi, const bool isAbsolute, const double minimum_peak_height,
+    GetDetectorsOffset::PeakLinearFunction &fit_result, bool &mask_it) {
+  // check y
+  double offset(0);
+  mask_it = false;
+
+  if (minimum_peak_height != EMPTY_DBL() &&
```

Comment:
`this->isEmpty()` is the correct check to use. Also, if the default value of the parameter was zero, this extra check could go away.